File: app/src/main/java/mx/dev/franco/automusictagfixer/di/StorageModule.java
package mx.dev.franco.automusictagfixer.di;

import android.app.Application;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import mx.dev.franco.automusictagfixer.fixer.AudioTagger;

@Module
public class StorageModule {

    @Singleton
    @Provides
    AudioTagger.StorageHelper providesStorageHelper(Application application) {
        return AudioTagger.StorageHelper.getInstance(application);
    }
}

File: app/src/main/java/mx/dev/franco/automusictagfixer/di/StorageModule.kt
package mx.dev.franco.automusictagfixer.di

import android.content.Context
import dagger.Module
import dagger.Provides
import mx.dev.franco.automusictagfixer.fixer.StorageHelper
import javax.inject.Singleton

@Module
class StorageModule {
    @Singleton
    @Provides
    fun providesStorageHelper(application: Context): StorageHelper {
        return StorageHelper.getInstance(application)
    }
}